### DISCUSSION TITLE
Is there a way to programtically set @router.get parameters for an endpoint?

### DISCUSSION THREAD

**Discussion by be-wise-be-kind** [fastapi/fastapi#8808] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get(
    "/api/v1/experimental/productions/",
    tags=["productions"],
    response_model=List[Production],
    response_description="A list of productions matching the given filter criteria",
    responses={209: {"model": EchoResponse}},
)
@my_custom_decorator
async def get_productions(query: QmqlQuery) -> List[Production]:
    """Return a list of productions that match specified criteria."""
    ...
```

### Description

Love the tool - thanks for all the hard work ....

Q: Given the code above, note my custom decorator `my_custom_decorator`.  For that decorator to work, I need to add a response to the @router.get decorator, namely `responses={209: {"model": EchoResponse}}`. Since my decorator and the response must always co-exist, it makes sense to me to add the response from inside my decorator - programatically - to be more DRY. Is there a prescribed mechanism for adding values to an existing response? 

### Operating System

Linux

### Operating System Details

Ubuntu

### FastAPI Version

0.68.0

### Python Version

3.8

**Comment by ComicShrimp** [comment]
You could try to use a middleware: https://fastapi.tiangolo.com/tutorial/middleware/
You can use the examples and "override" de response the way you want.
